src/main/java/org/opensearch/knn/index/codec/native_index/NativeIndexBuilderScratchIter.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.native_index;
+
+import java.io.IOException;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.Map;
+
+import org.apache.lucene.index.BinaryDocValues;
+import org.opensearch.knn.index.VectorDataType;
+import org.opensearch.knn.index.codec.util.KNNCodecUtil;
+import org.opensearch.knn.index.util.KNNEngine;
+import org.opensearch.knn.jni.JNICommons;
+import org.opensearch.knn.jni.JNIService;
+
+public class NativeIndexBuilderScratchIter extends NativeIndexBuilderScratch {
+
+    @Override
+    protected void createIndex(NativeIndexInfo indexInfo, BinaryDocValues values) throws IOException {
+        long indexAddress = initIndexFromScratch(
+            indexInfo.numDocs,
+            indexInfo.vectorInfo.dimension,
+            indexInfo.knnEngine,
+            indexInfo.parameters
+        );
+        KNNCodecUtil.VectorBatch batch = KNNCodecUtil.getVectorBatch(values, getVectorTransfer(indexInfo.vectorInfo.vectorDataType), true);
+        try {
+            for (; !batch.finished; batch = KNNCodecUtil.getVectorBatch(
```

Comment:
```suggestion
            for (; batch.finished == false; batch = KNNCodecUtil.getVectorBatch(
```

Reply:
I get this would be more readable, but isn't it convention to avoid == false?

Reply:
Resolved